hypothesis-python/tests/nocover/test_type_lookup.py

```diff
@@ -82,3 +83,23 @@ def test_callable_return_typegard_type():
 
     with pytest.raises(InvalidArgument, match="Cannot register generic type"):
         st.register_type_strategy(Callable[[], TypeGuard[int]], st.none())
+
+
+@dataclass
+class ItWorks:
+    a: dict[Tuple[int, int], str]
+
+
+@dataclass
+class ItDoesnt:
```

Comment:
Let's rename this now that it works, or at least leave a comment pointing back to the issue.